rpc: Validate provided keys for query_options parameter in listunspent

PastaPastaPasta opened the pull request:
At Dash, one of our developers was working with the `listunspent` RPC command, but instead of saying "minimumAmount" he said "minimmumAmount" as such the RPC wasn't working as expected. 

In https://github.com/dashpay/dash/pull/3507 we implemented a check so that `listunspent` returns an error if an unrecognized option is given. I figured I might as well adapt the code and throw up a PR here.

Cheers!

luke-jr reviewed (changes requested):
Can you refactor this to use `RPCTypeCheckObj`?

PastaPastaPasta committed:
rpc: Validate provided keys for query_options parameter in listunspent

With this change listunspent will throw an error if there is a wrong key
in the query_option object.

Signed-off-by: pasta <pasta@dashboost.org>

PastaPastaPasta commented:
I like that, makes it simpler, changed, and squashed/force-pushed

PastaPastaPasta commented:
Tested with
```
generatetoaddress 105 bcrt1qwq0cnj557dn070dd8l52dhkathpx083qlg5v4w
listunspent 1 99999999 [] true "{\"minimumAmount\" : 49}"
listunspent 1 99999999 [] true "{\"minimumAmount\" : 51}"
listunspent 1 99999999 [] true "{\"minimumAAmount\" : 51}"
```

Everything worked as expected

PastaPastaPasta commented:
Resolved test failure. made amounts not have their type checked since that gets checked anyway in `AmountFromValue`

adaminsky commented:
ACK `a99a3c0bd`
Built, ran unit tests and functional tests, and manually tested RPCs with different `query_options` objects including unsupported parameters. Reviewed the code and it looks good.

meshcollider reviewed (commented):
Seems fine to me. utACK a99a3c0bd6d5476503c015e23be569295fdd190c

fanquake commented:
@MarcoFalke can you take a look here?

MarcoFalke commented:
RPCman will do this in the future, but nothing wrong adding something like that in the meantime
